libs/features/mas/src/aem-fragment.js

```diff
@@ -152,47 +174,53 @@ export class AemFragment extends HTMLElement {
             .then(() => {
                 this.dispatchEvent(
                     new CustomEvent(EVENT_AEM_LOAD, {
-                        detail: this.data,
+                        detail: { ...this.data, stale: this.#stale },
                         bubbles: true,
                         composed: true,
                     }),
                 );
                 return true;
             })
             .catch((e) => {
-                /* c8 ignore next 3 */
-                this.#fail('Network error: failed to load fragment');
+                if (this.#rawData) {
+                    cache.add(this.#rawData);
+                    return true;
+                }
                 this.#readyPromise = null;
+                this.#fail(e);
                 return false;
             });
-        this.#readyPromise;
     }
 
-    #fail(error) {
+    #fail({ message, context }) {
         this.classList.add('error');
+        this.#log.error(`aem-fragment: ${message}`, context);
         this.dispatchEvent(
             new CustomEvent(EVENT_AEM_ERROR, {
-                detail: error,
+                detail: { message, ...context },
                 bubbles: true,
                 composed: true,
             }),
         );
     }
 
     async fetchData() {
-        this.#rawData = null;
-        this.#data = null;
+        this.classList.remove('error');
         let fragment = cache.get(this.#fragmentId);
-        if (!fragment) {
-            fragment = await getFragmentById(
-                baseUrl,
-                this.#fragmentId,
-                this.#author,
-                this.#ims ? headers : undefined,
-            );
-            cache.add(fragment);
+        if (fragment) {
+            this.#rawData = fragment;
+            return;
         }
+        this.#stale = true;
+        fragment = await getFragmentById(
+            baseUrl,
+            this.#fragmentId,
+            this.#author,
+            this.#ims ? headers : undefined,
+        );
+        cache.add(fragment);
         this.#rawData = fragment;
+        this.#stale = false;
```

Comment:
and here add a marker too?

Reply:
the marker is in `getFragmentById` right after. In between there is almost nothing.